src/main/java/org/opensha/commons/param/impl/ParameterListParameter.java

```
@@ -51,118 +50,51 @@ public ParameterListParameter(String name) {
+	public void setValue(ParameterList value) throws ParameterException {
+		// ListIterator it = value.getParametersIterator();
+		super.setValue(value);
+		// Setting the independent Param List for this parameter
 		this.setIndependentParameters(value);
 	}
 
-	/**
-	 * Compares value to see if equal.
-	 *
-	 * @param  obj                     The object to compare this to
-	 * @return                         True if the values are identical
-	 * @exception  ClassCastException  Is thrown if the comparing object is not
-	 *      a ParameterListParameter.
-	 */
-//	@Override
-//	public boolean equals(Object obj) {
-//		String S = C + ":equals(): ";
-//
-//		if (! (obj instanceof ParameterListParameter)) {
-//			throw new ClassCastException(S +
-//			"Object not a ParameterListParameter, unable to compare");
-//		}
-//
-//		String otherName = ( (ParameterListParameter) obj).getName();
-//		if ( (compareTo(obj) == 0) && getName().equals(otherName)) {
-//			return true;
-//		}
-//		else {
-//			return false;
-//		}
-		
-//		if (this == obj) return true;
-//		if (!(obj instanceof ParameterListParameter)) return false;
-//		ParameterListParameter plp = (ParameterListParameter) obj;
-//		return (value.equals(plp.getValue()) && getName().equals(plp.getName()));
-//
-//	}
-
 	/**
 	 *  Returns a copy so you can't edit or damage the origial.
 	 *
 	 * @return    Exact copy of this object's state
 	 */
-	public Object clone(){
-
+	public Object clone() {
 		ParameterListParameter param = null;
-		if( value == null ) param = new ParameterListParameter( name);
-		else param = new ParameterListParameter(name,(ParameterList)value);
-		if( param == null ) return null;
-		param.editable = true;
+		try {
+			if (value == null)
+				param = new ParameterListParameter(name);
+			else
+				param = new ParameterListParameter(name,(ParameterList)value);
+			param.editable = true;
+		} catch(Exception e) {
+			e.printStackTrace();
+			param.editable = false;
```

What's the motivation for this?

---

In both cases, param should be assigned to a new ParameterListParameter, so there's no case where it would be null unless construction failed (i.e. ConstraintException thrown in AbstractParameter).

Instead of silently returning null, this catches and prints the exception and reflects the fact that the param can't be edited as it's null.

---

Did you encounter it returning null?

The way I read the old code, the `if( param == null ) return null;` could never hit because `param` was assigned by either clause of the if/else, and there was no try/catch.

Your revised code I think is trying to return null if there's an exception, but it feels better to me for the exception to just be propagated to the caller as it was before. Also, I think your try/catch would introduce an NPE because you call `param.editable = false;` in a try/catch that can only be triggered if there's an instantiation error on the  `ParameterListParameter` constructor. In other words, the only way to get into the catch block results in `param` being null, and then you assign a variable inside `param` which would throw an NPE.

Unless I'm missing something, I would just remove the `if( param == null ) return null;` clause that can't be hit in the original method and otherwise keep as is.